pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -133,12 +132,17 @@ func (a *acrProvider) Provide() credentialprovider.DockerConfig {
 		return cfg
 	}
 	for ix := range *res.Value {
-		// TODO: I don't think this will work for national clouds
-		cfg[fmt.Sprintf("%s.azurecr.io", *(*res.Value)[ix].Name)] = entry
+		loginServer := getLoginServer((*res.Value)[ix])
+		glog.V(4).Infof("Adding Azure Container Registry docker credential for %s", loginServer)
+		cfg[loginServer] = entry
 	}
 	return cfg
 }
 
+func getLoginServer(registry containerregistry.Registry) string {
+	return *(*registry.RegistryProperties).LoginServer
```

Comment:
is there always a default? does it need a fallback to old behavior if it's empty?